src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -127,6 +130,28 @@ private static void AssertSetStrictlyEqual(string[] expected, string[] actual)
             Assert.False(sourceNamedType.IsImplicitlyDeclared);
         }
 
+        if (type is PENamedTypeSymbol peType)
+        {
+            var module = (PEModuleSymbol)type.ContainingModule;
+            var reader = module.Module.GetMetadataReader();
+            var fieldDefHandle = reader.GetTypeDefinition(peType.Handle).GetFields()
+                .Where(f => reader.GetString(reader.GetFieldDefinition(f).Name) == WellKnownMemberNames.ExtensionFieldName).SingleOrDefault();
+
+            // Static extensions don't have this field, but non-static extensions have it
+            Assert.Equal(fieldDefHandle.IsNil, type.IsStatic);
+
+            if (!type.IsStatic)
+            {
+                var fieldDef = reader.GetFieldDefinition(fieldDefHandle);
+                var blob = reader.GetBlobReader(fieldDef.Signature);
+                var decoder = new SignatureDecoder<string, object>(ConstantSignatureVisualizer.Instance, reader, genericContext: null);
+                var fieldTypeDisplay = decoder.DecodeFieldSignature(ref blob);
+
+                // The instance value field has the expected type
+                Assert.Equal(fieldType, fieldTypeDisplay);
```

Comment:
>Assert.Equal(fieldType, fieldTypeDisplay); [](http://example.com/codeflow?start=16&length=42)

It feels somewhat surprising that validation on import is more thorough than in the tests.

Reply:
> I didn't follow. What do you mean?

I mean ```validateUnderlyingInstanceField``` function checks more things than this code.

Reply:
Added missing checks